packages/api/cms-api/src/common/guards/external-request-without-header.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { Request } from "express";
+
+@Injectable()
+export class ExternalRequestWithoutHeaderGuard implements CanActivate {
+    private headerName: string;
+    private headerValue: string;
+
+    constructor({ headerName, headerValue }: { headerName: string; headerValue: string }) {
+        this.headerName = headerName;
+        this.headerValue = headerValue;
+    }
+
+    private isRequestInternal(request: Request): boolean {
+        // This check is only ok when the application is behind a trusted proxy
+        return !request.header("x-forwarded-for");
```

Comment:
How is this secure? Couldn't anyone add the header to an API request? I suppose the CDN takes care of removing this header from incoming requests? 

Is there any scenario where the API can still be reached directly bypassing the CDN?